src/components/diff-settings.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+/**
+ * @typedef DiffSettingsProps
+ * @property {object} diffSettings
+ * @property {string} diffType
+ * @property {Function} handleRemoveFormattingChange
+ */
+
+/**
+ *
+ * @class DiffSettings
+ * @extends {React.PureComponent}
+ * @param {DiffSettingsProps} props
+
+ */
+export default class DiffSettings extends React.PureComponent {
+  constructor (props) {
+    super(props);
+
+    this._handleRemoveFormattingChange = this._handleRemoveFormattingChange.bind(this);
+  }
+
+  render () {
+    const renderedDiffTypes = ['SIDE_BY_SIDE_RENDERED', 'HIGHLIGHTED_RENDERED'];
+    if (!renderedDiffTypes.includes(this.props.diffType)) {
+      return null;
+    }
+
+    return (
+      <label className="utilities__label">
+        <input
+          checked={this.props.diffSettings.removeFormatting}
+          className="utilities__input"
+          onChange={this._handleRemoveFormattingChange}
+          type="checkbox">
+        </input>
+        Remove formatting
+      </label>
+    );
+  }
+
+  _handleRemoveFormattingChange (event) {
+    this.props.handleRemoveFormattingChange(event.target.checked);
```

Comment:
Can this please return an object with `removeFormatting` as a property? That way it matches up with the props you are passing in. My whole point with this setup was that this view encapsulates a potentially huge *variety* of settings for any given diff. That this is “the form for setting**s** on diffs,” not “the form for the removeFormatting setting.” (Obviously that is the only setting right now, but I really don’t think it’s likely to stay that way. Or at least I’d like to make it not a pain in the butt to change that.)